### PR TITLE
Fix: barBorderRadius didn't work

### LINKED ISSUE #12321: Bar backgroundStyle barBorderRadius invalid
### Version
4.7.0

### Steps to reproduce
{ backgroundStyle: {barBorderRadius: 20} } 

### What is expected?
the bar background border radius

### What is actually happening?
invalid

### PR DESCRIPTION
**Pull request by susiwen8**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

Fix barBorderRadius didn't work, not only background but also item

### Fixed issues
Close #12321 

## Details
Screen Shot 2020-03-25 at 22 33 11

### After: How is it fixed in this PR?
Screen Shot 2020-03-25 at 22 32 32

### Are there any API changes?

- [ ] The API has been changed.

### Related test cases or examples to use the new APIs

NA.

### Merging options

- [ ] Please squash the commits into a single one when merge.

### CODE REVIEW

File: src/layout/barGrid.js [hunk 1]
```diff
@@ -478,7 +478,9 @@ export function layout(seriesType, ecModel) {
                 x: x,
                 y: y,
                 width: width,
-                height: height
+                height: height,
+                r: seriesModel.get('barBorderRadius'),
```
Comment (pissang) [inline src/layout/barGrid.js]: I believe this should be `itemModel.get('barBorderRadius')`?

Comment (susiwen8) [reply src/layout/barGrid.js]: @pissang item border radius wasn't set from here. so I have it removed.

File: src/chart/bar/BarView.js [hunk 2]
```diff
@@ -440,6 +441,7 @@ var getLayout = {
         return {
             x: layout.x + signX * fixedLineWidth / 2,
             y: layout.y + signY * fixedLineWidth / 2,
+            r: layout.rb,
             width: layout.width - signX * fixedLineWidth,
```
Comment (pissang) [inline src/chart/bar/BarView.js]: This layout is used for bar rect element. `r` property here will not be used.

File: src/chart/bar/BarView.js [hunk 3]
```diff
@@ -673,6 +675,7 @@ function createBackgroundShape(isHorizontalOrRadial, layout, coord) {
         return {
             x: isHorizontalOrRadial ? layout.x : coordLayout.x,
             y: isHorizontalOrRadial ? coordLayout.y : layout.y,
+            r: isHorizontalOrRadial ? layout.r : coordLayout.r,
             width: isHorizontalOrRadial ? layout.width : coordLayout.width,
```
Comment (pissang) [inline src/chart/bar/BarView.js]: I think it should be `layout.rb` here. Also coordLayout is a rect. It don't have r property.

Also, I prefer not to put `radius` in the layout. See comment below.

File: src/layout/barGrid.js [hunk 4]
```diff
@@ -478,7 +478,8 @@ export function layout(seriesType, ecModel) {
                 x: x,
                 y: y,
                 width: width,
-                height: height
+                height: height,
+                rb: seriesModel.get('backgroundStyle.barBorderRadius')
             });
```
Comment (pissang) [inline src/layout/barGrid.js]: `radius` is not a layout related property. So I think it's better to set the property here.

https://github.com/apache/incubator-echarts/blob/b4b979a2fb9ad97493e7efb7c4d64055325ad214/src/chart/bar/BarView.js#L146

```js
bgEl.setShape('r', backgroundModel.get('barBorderRadius'));
```

**Pull request merged**